ethereum/core/src/main/java/org/hyperledger/besu/ethereum/vm/operations/BlockHashOperation.java

```diff
@@ -15,55 +15,70 @@
-    super(0x40, "BLOCKHASH", 1, 1, false, 1, gasCalculator);
+    super(
+        0x40,
+        "BLOCKHASH",
+        1,
+        1,
+        false,
+        1,
+        gasCalculator,
+        gasCalculator.getBlockHashOperationGasCost());
   }
 
   @Override
-  public Gas cost(final MessageFrame frame) {
-    return gasCalculator().getBlockHashOperationGasCost();
-  }
+  public OperationResult execute(final MessageFrame frame, final EVM evm) {
+    try {
+      if (frame.getRemainingGas().compareTo(gasCost) < 0) {
+        return oogResponse;
+      }
+      final UInt256 blockArg = UInt256.fromBytes(frame.popStackItem());
 
-  @Override
-  public void execute(final MessageFrame frame) {
-    final UInt256 blockArg = UInt256.fromBytes(frame.popStackItem());
+      // Short-circuit if value is unreasonably large
+      if (!blockArg.fitsLong()) {
+        frame.pushStackItem(Bytes32.ZERO);
+        return successResponse;
+      }
 
-    // Short-circuit if value is unreasonably large
-    if (!blockArg.fitsLong()) {
-      frame.pushStackItem(Bytes32.ZERO);
-      return;
-    }
+      final long soughtBlock = blockArg.toLong();
+      final ProcessableBlockHeader blockHeader = frame.getBlockHeader();
+      final long currentBlockNumber = blockHeader.getNumber();
+      final long mostRecentBlockNumber = currentBlockNumber - 1;
 
-    final long soughtBlock = blockArg.toLong();
-    final ProcessableBlockHeader blockHeader = frame.getBlockHeader();
-    final long currentBlockNumber = blockHeader.getNumber();
-    final long mostRecentBlockNumber = currentBlockNumber - 1;
+      // If the current block is the genesis block or the sought block is
+      // not within the last 256 completed blocks, zero is returned.
+      if (currentBlockNumber == BlockHeader.GENESIS_BLOCK_NUMBER
+          || soughtBlock < (mostRecentBlockNumber - MAX_RELATIVE_BLOCK)
+          || soughtBlock > mostRecentBlockNumber) {
+        frame.pushStackItem(Bytes32.ZERO);
+      } else {
+        final BlockHashLookup blockHashLookup = frame.getBlockHashLookup();
+        final Hash blockHash = blockHashLookup.getBlockHash(soughtBlock);
+        frame.pushStackItem(blockHash);
+      }
 
-    // If the current block is the genesis block or the sought block is
-    // not within the last 256 completed blocks, zero is returned.
-    if (currentBlockNumber == BlockHeader.GENESIS_BLOCK_NUMBER
-        || soughtBlock < (mostRecentBlockNumber - MAX_RELATIVE_BLOCK)
-        || soughtBlock > mostRecentBlockNumber) {
-      frame.pushStackItem(Bytes32.ZERO);
-    } else {
-      final BlockHashLookup blockHashLookup = frame.getBlockHashLookup();
-      final Hash blockHash = blockHashLookup.getBlockHash(soughtBlock);
-      frame.pushStackItem(blockHash);
+      return successResponse;
+    } catch (final UnderflowException ue) {
+      return UNDERFLOW_RESPONSE;
+    } catch (final OverflowException oe) {
+      return OVERFLOWFLOW_RESPONSE;
```

Comment:
`OVERFLOWFLOW_RESPONSE`? Do you want to say `OVERFLOW_RESPONSE`

Reply:
Yep.  Fix pushed.